resqpy/well/_blocked_well.py

```diff
@@ -2350,64 +2349,94 @@ def __add_as_properties(self, df, add_as_properties, extra_columns_list, row_ci_
             pk, uom = self.__set_pk_and_uom_for_df_properties(extra = extra, length_uom = length_uom)
             # 'SKIN': use defaults for now; todo: create local property kind for skin
-            expanded = np.full(self.cell_count, np.NaN)
-            expanded[ci_map] = df[extra]
+            expanded = np.append(df[column].to_numpy(), missing_cells_nan_array)
             extra_pc.add_cached_array_to_imported_list(expanded,
                                                        'blocked well dataframe',
                                                        extra,
-                                                       discrete = False,
+                                                       discrete = (extra == "STAT"),
                                                        uom = uom,
                                                        property_kind = pk,
                                                        local_property_kind_uuid = None,
                                                        facet_type = None,
                                                        facet = None,
                                                        realization = None,
                                                        indexable_element = 'cells',
-                                                       count = 1)
+                                                       count = 1,
+                                                       time_index = time_index)
         extra_pc.write_hdf5_for_imported_list()
-        extra_pc.create_xml_for_imported_list_and_add_parts_to_model()
+        extra_pc.create_xml_for_imported_list_and_add_parts_to_model(time_series_uuid = time_series_uuid)
 
-    def __set_pk_and_uom_for_df_properties(self, extra, length_uom):
+    def __set_pk_and_uom_for_df_properties(self, extra, length_uom, temperature_uom = None):
         """Set the property kind and unit of measure for all properties in the dataframe."""
-        column_list = ['ANGLA', 'ANGLV', 'KH', 'PPERF']
-        uom_list = ['dega', 'dega', 'mD.' + length_uom, length_uom + '/' + length_uom]
-        pk_list = ['azimuth', 'inclination', 'permeability_length', 'continuous'
-                  ]  # neither azimuth nor dip are correct property kinds; todo: create local property kinds
-        if extra in column_list:
-            list_position = column_list.index(extra)
-            uom, pk = uom_list[list_position], pk_list[list_position]
-        elif extra in ['LENGTH', 'MD', 'X', 'Y', 'DEPTH', 'RADW']:
-            uom, pk = self.__set_pk_and_uom_for_length_based_properties(length_uom = length_uom, extra = extra)
-        else:
-            uom = 'Euc'
-            pk = 'continuous'
+        if length_uom not in ['m', 'ft']:
+            raise ValueError(f"The length_uom {length_uom} must be either 'm' or 'ft'.")
+        if extra == 'TEMP' and (temperature_uom is None or temperature_uom[0].upper() not in ['C', 'F']):
```

Comment:
I think we should be using RESQML uoms here
<img width="546" alt="Screenshot 2022-07-18 at 12 47 04" src="https://user-images.githubusercontent.com/71271320/179496060-10620d6c-21e3-4952-b227-bfbe582d9e72.png">
{'K', 'degC', 'degF', 'degR'}: